### PR TITLE
[stable/elasticsearch] Add ServiceAccounts for each components + PodSecurityPolicy

### PR DESCRIPTION
**Pull request by spiarh**
Signed-off-by: Ludovic Cavajani <lcavajani@suse.com>

#### What this PR does / why we need it:
This PR allows one to create ServiceAccount to run each component instead of the default ServiceAccount.

This is essential for my future PR which enables PodSecurityPolicy.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

File: stable/elasticsearch/templates/client-deployment.yaml [hunk 1]
```diff
@@ -21,6 +21,7 @@ spec:
 {{ toYaml .Values.client.podAnnotations | indent 8 }}
         {{- end }}
     spec:
+      serviceAccountName: {{ template "elasticsearch.serviceAccountName.client" . }}
```
Comment (rendhalver) [inline stable/elasticsearch/templates/client-deployment.yaml:24]: Why does this not use `template "elasticsearch.client.fullname"`?
Or alternatively why does the name of the client ServiceAccount not use this name?
Should they not be consistent?

**Review by rendhalver — CHANGES_REQUESTED** [review]

File: stable/elasticsearch/templates/client-deployment.yaml (hunk 1 above)
Comment (spiarh) [reply stable/elasticsearch/templates/client-deployment.yaml:24]: With this, we can:

1. default to the "default"  ServiceAccount if create is false and name is empty.
2. or use specified ServiceAccount if create is false but name has a value.

Comment (rendhalver) [reply stable/elasticsearch/templates/client-deployment.yaml:24]: Oh I see what you are doing.

**Comment by rendhalver** [comment]
Why not put your PodSecurityPolicy into this PR?

**Comment by lcavajani** [comment]
if that's ok for you, I'll add it today ;)

File: stable/elasticsearch/templates/podsecuritypolicy.yaml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+  labels:
+    app: {{ template "elasticsearch.name" . }}
+    chart: {{ .Chart.Name }}-{{ .Chart.Version }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+  annotations:
+    seccomp.security.alpha.kubernetes.io/allowedProfileNames: '*'
+    seccomp.security.alpha.kubernetes.io/defaultProfileName: 'docker/default'
+    apparmor.security.beta.kubernetes.io/defaultProfileName: 'runtime/default'
+spec:
```
Comment (rendhalver) [inline stable/elasticsearch/templates/podsecuritypolicy.yaml:15]: Do these privileges need to be configurable?

Comment (spiarh) [reply stable/elasticsearch/templates/podsecuritypolicy.yaml:15]: This should not, it just applies some default but they can still be overridden with annotations.

Comment (rendhalver) [reply stable/elasticsearch/templates/podsecuritypolicy.yaml:15]: Ok cool.

Comment (spiarh) [reply stable/elasticsearch/templates/podsecuritypolicy.yaml:15]: I will make these optional in case apparmor is not enabled for example

**Pull request merged**